https://bugs.gentoo.org/903001
https://github.com/libssh2/libssh2/pull/700
https://github.com/libssh2/libssh2/commit/b952674f120748174ed2c0fb93e7bd78cf355cac

From f0681a4573d7c7f7484d3157ddff7063a200295b Mon Sep 17 00:00:00 2001
From: Viktor Szakats <commit@vsz.me>
Date: Thu, 19 May 2022 13:25:06 +0000
Subject: [PATCH] openssl: add support for LibreSSL 3.5.x

LibreSSL 3.5.0 made more structures opaque, so let's enable existing
support for that when building against these LibreSSL versions.

Ref: https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.5.0-relnotes.txt
---
 src/openssl.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/openssl.h b/src/openssl.h
index 658b040..09bfd7c 100644
--- a/src/openssl.h
+++ b/src/openssl.h
@@ -57,8 +57,9 @@
 #include <openssl/pem.h>
 #include <openssl/rand.h>
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
-    !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)) || \
+    LIBRESSL_VERSION_NUMBER >= 0x3050000fL
 # define HAVE_OPAQUE_STRUCTS 1
 #endif
 
